src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/PlayerDeathSystem.java

```diff
@@ -94,7 +103,9 @@ private void dropItemsFromInventory(EntityRef player) {
         for (int slotNumber = 0; slotNumber <= inventorySize; slotNumber++) {
             EntityRef slot = inventoryManager.getItemInSlot(player, slotNumber);
             Prefab currentPrefab = slot.getParentPrefab();
-            if (currentPrefab != null && !currentPrefab.equals(staffPrefab)) {
+            if (currentPrefab != null && currentPrefab.equals(staffPrefab)) {
+                inventoryManager.removeItem(player, EntityRef.NULL, slotNumber, true, 1);
```

Comment:
I don't think we should treat the staff different than every other item in the inventory here.
I'm aware that there is a task where we will need to discuss which items to treat how (keep / drop / remove from the game), but I think when we do that we should do it for all items and not start doing it for the staff already now.
Hence, I would drop the staff on death like all the other items.

Reply:
I didn't want to remove the logic implemented previously, wherein the staff is not dropped on death, but now as we are giving the staff too on respawn I removed it every time the player dies. I could remove this in case the previously implemented logic is not what is required.

Reply:
I think until we think it through properly, which items should be kept, dropped, or removed, we should not have additional logic treating some different than others. For me this includes the logic that already was there. Thus, at least we have consistent behaviour for all items.